File: 0788-rotated-digits/0788-rotated-digits.py
class Solution:
    def rotatedDigits(self, n):
        ans = 0
        for i in range(n+1):
            if self.helper(i):
                ans += 1
        return ans
    
    def helper(self, x):
        flag = False
        while x > 0:
            t = x % 10
            x //= 10
            if t == 4 or t == 3 or t == 7:
                return False
            if t == 6 or t == 9 or t == 2 or t == 5:
                flag = True
        return flag

File: 0788-rotated-digits/0788-rotated-digits.java
class Solution {
    public int rotatedDigits(int n) {
        int ans = 0;
        for(int i = 0; i <= n; ++i) {
            if(helper(i)) ++ans;
        }
        return ans;
    }
    
    Boolean helper(int x) {
        boolean flag = false;
        while(x > 0) {
            int t = x % 10;
            x /= 10;
            if(t == 4 || t == 3 || t == 7) return false;
            if(t == 6 || t == 9 || t == 2 || t == 5) flag = true;
        }
        return flag;
    }
}